src/EditorFeatures/Test2/IntelliSense/CSharpCompletionCommandHandlerTests.vb

```diff
@@ -5996,6 +5996,35 @@ namespace NS
             End Using
         End Function
 
+        <WorkItem(39519, "https://github.com/dotnet/roslyn/issues/39519")>
+        <WpfFact, Trait(Traits.Feature, Traits.Features.Completion)>
+        Public Async Function TestSuggestedNamesDontStartWithDigit() As Task
+            Using state = TestStateFactory.CreateCSharpTestState(
+                  <Document><![CDATA[
+namespace NS
+{
+    class C
+    {
+        public void Foo(Foo123Bar $$)
+        {
+        }
+    }
+
+    public class Foo123Bar
+    {
+    } 
+}
+]]></Document>)
+                Dim workspace = state.Workspace
+                workspace.TryApplyChanges(workspace.CurrentSolution.WithOptions(workspace.Options _
+                    .WithChangedOption(CompletionOptions.ShowNameSuggestions, LanguageNames.CSharp, True)))
+
+                state.SendInvokeCompletionList()
+                Await state.AssertCompletionItemsContainAll("foo123Bar", "foo123", "foo", "bar")
+                Await state.AssertCompletionItemsDoNotContainAny("123")
```

Comment:
```suggestion
                Await state.AssertCompletionItemsDoNotContainAny("123Bar")
```

Reply:
Take "FooCatBar" for example, we don't recommend "cat" in this case because it's not a left-most or right-most sub-array of words.

We should be checking for the absence of is "123Bar" since "123" wouldn't be offered even if it were a valid identifier. In the original issue, they checked against the number itself because the number was the rightmost "word" and therefore eligible. Try your test case out in VS before your change, and you'll see "123Bar" offered and not "123".

Reply:
Perhaps there should be a second test which is an exact copy of the original issue? That way we have this case where the number is in the middle (and part of a larger possible identifier), and one where the number itself becomes the possible identifier.

Reply:
Done.